### PR TITLE
Idva6 1142 add mappers

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

Added all of the mappers. This should unblock: 
- GET /acsp-members/acsps/{acsp_number}
- GET /acsp-members
- GET /acsp-members/{id}

__JIRA Ticket Number__

IDVA6-1142

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [x] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembersMapper.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    public AcspMembers daoToDto( final Page<AcspMembersDao> acspMembersDaos, final AcspDataDao acspDataDao ){
+        final var dtos =
+        acspMembersDaos.map( baseMapper::daoToDto )
+                       .map( mapperUtil::enrichAcspMembershipWithUserDetails )
+                       .map( dto -> {
+                           dto.setAcspName( acspDataDao.getAcspName() );
+                           dto.setAcspStatus( AcspStatusEnum.fromValue( acspDataDao.getAcspStatus() ) );
+                           return dto;
+                       } );
```
Comment (tferns-ch) [inline src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembersMapper.java:32]: Isn't it worth doing all of this in a single map? 

```suggestion
        final var dtos =
        acspMembersDaos.map(dao -> {
            var dto = baseMapper.daoToDto(dao);
            dto = mapperUtil.enrichAcspMembershipWithUserDetails(dto);
            dto.setAcspName(acspDataDao.getAcspName());
            dto.setAcspStatus(AcspStatusEnum.fromValue(acspDataDao.getAcspStatus()));
            return dto;
        });
```

**Review by tferns-ch — APPROVED** [review]
Nice one

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipMapper.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private final BaseMapper baseMapper;
+
+    private final MapperUtil mapperUtil;
+
+    public AcspMembershipMapper( final BaseMapper baseMapper, final MapperUtil mapperUtil ) {
+        this.baseMapper = baseMapper;
+        this.mapperUtil = mapperUtil;
+    }
+
+    public AcspMembership daoToDto( final AcspMembersDao acspMembersDao ){
```
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipMapper.java:21]: do we need this mapper?

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembersMapper.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.baseMapper = baseMapper;
+        this.mapperUtil = mapperUtil;
+    }
+
+    public AcspMembers daoToDto( final Page<AcspMembersDao> acspMembersDaos, final AcspDataDao acspDataDao ){
+        final var dtos =
+        acspMembersDaos.map( baseMapper::daoToDto )
+                       .map( mapperUtil::enrichAcspMembershipWithUserDetails )
+                       .map( dto -> {
+                           dto.setAcspName( acspDataDao.getAcspName() );
```
[the remark this answers is not in the document]
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembersMapper.java:29]: acsp_number is already covered by the baseMapper

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipListMapper.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        this.baseMapper = baseMapper;
+        this.mapperUtil = mapperUtil;
+    }
+
+    public List<AcspMembership> daoToDto( final List<AcspMembersDao> acspMembersDaos, final User userData ){
+        return acspMembersDaos.stream()
+                .map( baseMapper::daoToDto )
+                .map( mapperUtil::enrichAcspMembershipWithAcspData )
+                .map( dto -> {
+                    dto.setUserEmail( userData.getEmail() );
```
[the remark this answers is not in the document]
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipListMapper.java:31]: user_id is already covered by baseMapper

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipMapper.java (hunk 2 above)
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/mapper/AcspMembershipMapper.java:21]: Yeah - it's for Suvarna's endpoint

**Review by kkonuganti-ch — APPROVED** [review]

**Pull request merged**